Mutagen.Bethesda.Analyzers.Skyrim/Record/Dialog/Responses/SkillCheckAnalyzer.cs

```diff
@@ -0,0 +1,57 @@
+﻿using Mutagen.Bethesda.Analyzers.SDK.Analyzers;
+using Mutagen.Bethesda.Analyzers.SDK.Results;
+using Mutagen.Bethesda.Analyzers.SDK.Topics;
+using Mutagen.Bethesda.Skyrim;
+namespace Mutagen.Bethesda.Analyzers.Skyrim.Record.Dialog.Responses;
+
+public class SkillCheckAnalyzer : IIsolatedRecordAnalyzer<IDialogResponsesGetter>
+{
+    public static readonly TopicDefinition<string> NonPlayerSkillCheck = MutagenTopicBuilder.DevelopmentTopic(
+            "Non-Player Skill Check",
+            Severity.Warning)
+        .WithFormatting<string>("Skill check in dialog are not checked on the player but on {0} - this is usually a sign of a mistake");
+
+    public static readonly TopicDefinition<string> NonGlobalSkillCheck = MutagenTopicBuilder.DevelopmentTopic(
+            "Non-Global Skill Check",
+            Severity.Suggestion)
+        .WithFormatting<string>("Skill check in dialog doesn't use global to evaluate skill level but {0} - this is usually a sign of a mistake");
```

Comment:
Any reason you marked on as warning, the other as a suggestion?

Reply:
The non-player skill check is usually an error that causes unexpected behavior but we're not sure so it's a warning. The global thing is demanded by convention but technically it also works when it's a float.